### PR TITLE
Cut 0.6.0; Reverted default values for metric types, let's not break compatibility there

### PR DESCRIPTION
**Pull request by bwplotka**

### CODE REVIEW

**Review by jmichalek132 — APPROVED** [review]
Small nit since you don't want to break combability with this release do we also want to revert the change from this pr #99?

LGTM otherwise.

File: cmd/avalanche/avalanche.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func main() {
 	tlsClientInsecure := kingpin.Flag("tls-client-insecure", "Skip certificate check on tls connection").Default("false").Bool()
 	remoteTenantHeader := kingpin.Flag("remote-tenant-header", "Tenant ID to include in remote_write send. The default, is the default tenant header expected by Cortex.").Default("X-Scope-OrgID").String()
 	// TODO(bwplotka): Make this a non-bool flag (e.g. out-of-order-min-time).
-	outOfOrder := kingpin.Flag("out-of-order", "Enable out-of-order timestamps in remote write requests").Default("true").Bool()
+	outOfOrder := kingpin.Flag("remote-out-of-order", "Enable out-of-order timestamps in remote write requests").Default("true").Bool()
```
Comment (kakkoyun) [inline cmd/avalanche/avalanche.go:78]: This is a breaking change. How strict are we about this? Shall we have a shim for `out-of-order` flag with a deprecation warning?

Comment (bwplotka) [reply cmd/avalanche/avalanche.go:78]: So I wanted to respond "thanks for being careful but we never released it" but actually I double-checked now and we did in 0.5.0 🙈  https://github.com/prometheus-community/avalanche/commit/8494cab4bb806fcc7e115cac6a7fb3eff169b227

Thanks! Will fix. Do you feel it's worth doing?

**Comment by bwplotka** [comment]
> Small nit since you don't want to break combability with this release do we also want to revert the change from this pr https://github.com/prometheus-community/avalanche/pull/99?

With any release we ideally don't break compatibility, but we want and can break compatibility (we are 0.x version) if necessary. I believe metric-interval is worth it, but I reverted metric types as it's more surprising and unwanted potentially. WDYT? @jmichalek132

**Pull request merged**